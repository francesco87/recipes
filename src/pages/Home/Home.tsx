import Search from '../../components/Search/Search.tsx';
import { Box } from '@mui/material';
import { Typography } from '@mui/material';

const Home = () => {
  return (
    <Box>
      <Typography variant="h2" gutterBottom>
        Best recipes
      </Typography>
      <Search />
    </Box>
  );
};

export default Home;
