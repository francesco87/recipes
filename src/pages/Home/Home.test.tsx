import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import Home from './Home';

vi.mock('../../components/Search/Search.tsx', () => ({
  __esModule: true,
  default: () => <div data-testid="search-component" />,
}));

describe('Home Component', () => {
  it('Home Snapshot', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it('Renders Home title', () => {
    render(<Home />);
    expect(screen.getByText('Best recipes')).toBeInTheDocument();
  });

  it('Renders Search component', () => {
    render(<Home />);
    expect(screen.getByTestId('search-component')).toBeInTheDocument();
  });
});
