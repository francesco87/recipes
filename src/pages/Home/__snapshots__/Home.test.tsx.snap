// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > Home Snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-1lcbsct-MuiTypography-root"
    >
      Best recipes
    </h2>
    <div
      data-testid="search-component"
    />
  </div>
</div>
`;
